DataManagementSystem/Utilities/MetaQuery.py

```diff
@@ -0,0 +1,208 @@
+########################################################################
+# File: MetaQuery.py
+# Author: A.T.
+# Date: 24.02.2015
+# $HeadID$
+########################################################################
+
+""" Utilities for managing metadata based queries
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC import S_OK, S_ERROR
+import DIRAC.Core.Utilities.Time as Time
+
+from types import ListType, DictType, StringTypes, IntType, LongType, FloatType
+import json
+
+class MetaQuery( object ):
+
+  def __init__( self, queryDict = None, metaTypeDict = None ):
+    self.__metaQueryDict = queryDict
+    self.__metaTypeDict = metaTypeDict
+
+  def setMetaQuery( self, queryList, metaTypeDict ):
+    """ Create the metadata query out of the command line arguments
+    """
+    self.__metaTypeDict = metaTypeDict
+    metaDict = {}
+    contMode = False
+    value = ''
+    for arg in queryList:
+      if not contMode:
+        operation = ''
+        for op in ['>=','<=','>','<','!=','=']:
+          if op in arg:
+            operation = op
+            break
+        if not operation:
+          return S_ERROR( 'Illegal query element %s' % arg )
+
+        name,value = arg.split(operation)
+        if not name in metaTypeDict:
+          return S_ERROR( "Metadata field %s not defined" % name )
+
+        mtype = metaTypeDict[name]
+      else:
+        value += ' ' + arg
+        value = value.replace(contMode,'')
+        contMode = False
+
+      if value[0] in ['"', "'"] and value[-1] not in ['"', "'"]:
+        contMode = value[0]
+        continue
+
+      if ',' in value:
+        valueList = [ x.replace("'","").replace('"','') for x in value.split(',') ]
+        mvalue = valueList
+        if mtype[0:3].lower() == 'int':
+          mvalue = [ int(x) for x in valueList if not x in ['Missing','Any'] ]
+          mvalue += [ x for x in valueList if x in ['Missing','Any'] ]
+        if mtype[0:5].lower() == 'float':
+          mvalue = [ float(x) for x in valueList if not x in ['Missing','Any'] ]
+          mvalue += [ x for x in valueList if x in ['Missing','Any'] ]
+        if operation == "=":
+          operation = 'in'
+        if operation == "!=":
+          operation = 'nin'
+        mvalue = {operation:mvalue}
+      else:
+        mvalue = value.replace("'","").replace('"','')
+        if not value in ['Missing','Any']:
+          if mtype[0:3].lower() == 'int':
+            mvalue = int(value)
+          if mtype[0:5].lower() == 'float':
+            mvalue = float(value)
+        if operation != '=':
+          mvalue = {operation:mvalue}
+
+      if name in metaDict:
+        if type(metaDict[name]) == DictType:
```

Comment:
All these kinds of checks should be rather done as:

 isinstance( metaDict[name], dict )